chaincode/main.go

```diff
@@ -189,17 +204,25 @@ func (t *SubstraChaincode) Invoke(stub shim.ChaincodeStubInterface) peer.Respons
 
 	// Invoke duration
 	duration := int(time.Since(start).Nanoseconds()) / 1e6
-
 	// Return the result as success payload
 	if err != nil {
-		logger.Errorf("[%s][%s] Response (%dms): '%#v' - Error: '%s'", stub.GetChannelID(), stub.GetTxID()[:10], duration, result, err)
+		logger.Errorf("[%s][%s] Response (%dms): '%#v','%s' - Error: '%s'", stub.GetChannelID(), stub.GetTxID()[:10], duration, result, bookmark, err)
 		return formatErrorResponse(err)
 	}
 
+	// Add bookmark (if any) to response
+	if hasBookmark {
+		result = map[string]interface{}{
+			"results":  result,
+			"bookmark": bookmark,
+		}
+	}
+
```

Comment:
I'm not a fan of this format but I understand the need. 
And I feel that this mechanic should be handle through interface for lists type output. But one again this is more complex and can come later.

Reply:
As it's only the first step of full pagination, we will see late if we can have a better format at the end :)